### DISCUSSION TITLE
User Data Protection guides

### DISCUSSION THREAD

**Discussion by tweakch** [cardano-foundation/developer-portal#164] [Ideas]
[GDPR](https://www.gdpr.eu) compliancy is a BIG issue for companies and contractors dealing with user data and user consent in and [outside](https://termly.io/resources/articles/gdpr-in-the-us/) of the EU.

There are similar regulations like [CCPA](https://en.m.wikipedia.org/wiki/California_Consumer_Privacy_Act) that touch on the same topics...

The way developers are usually confronted with these topics in projects is that there is a *bullet point* in the non-functional requirements section that says:

* GDPR compliant

Or, within functional requirements

* opt-in and opt-out of marketing newsletters

GDPR and regulations alike are much, much broader than just Opt-In and Opt-Out of course, but these "user choices" is what devs have to deal with. 

Over the past years, different schemas have emerged to capture consent. Details vary but these schemas look something like this:

```json
{
  "person":{...},
  "data":{...},
  "consents": [
    {
      "purpose": "marketing",
      "channels": [...],
      "timestamp": "123456789",
      "retention": "12345",
      "value":"optin"
    },
    ...
  ]
}
```

Implementing GDPR compliancy is a PAIN in the 🍑 because in addition to storing these consent records per channel, you MUST do tons of business logic with them (ie `value` and `retention`) and provide the data in human readable form upon request. (`SELECT * FROM 'gdpr.consents' t0 LEFT JOIN ... LEFT JOIN ... RIGHT JOIN ...` is enough... but I mean... ugh...) 

I'm in favour of data protection but I think they are being implemented in a wrong manner...

IMHO these consents should live on the blockchain and all I get is an `addr` where I can look them up and if I as a company act on these consents, I as a company am obliged to record what I used the consent for... Not *use it and report upon request*.

There are two (or three) scopes of interest here:

`User`, `Developer`, `Company`

The `user` wants to see and manage their consents in real time and not "upon request".

The `developer` wants to design websites and write code and not talk to legal about the wording of check box labels that give implicit over explicit consent.

Companies want to earn money and do whatever costs less to implement and maintain.

What do you think can we, as a developer portal, do to make life easier for `devs`, better and more protected for `users` and interesting or cost reducing for `companies`?

Additional resources
[GDPR Summary](https://docs.microsoft.com/en-us/compliance/regulatory/gdpr)
[Not supported in consent string](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/Consent%20string%20and%20vendor%20list%20formats%20v1.1%20Final.md#what-is-not-being-supported-in-the-consent-string-format-)
[Blockchain Privacy Poisoning](https://www.cpomagazine.com/data-protection/blockchain-privacy-poisoning-a-new-concern-in-post-gdpr-era)

**Comment by tweakch** [comment]
> What do you think can we, as a developer portal, do to make life easier for devs, better and more protected for users and interesting or cost reducing for companies? 

I would like @admin-cf thoughts on this?

**Comment by katomm** [comment]
Please note that the admin-cf account is a special account to manage things, not necessarily associated with a specific user behind it.

**Reply by tweakch** [reply]
I see, thank you for the info.
